File: leetcode.com/python/949_Largest_Time_for_Given_Digits.py
import itertools
class Solution(object):
    def largestTimeFromDigits(self, A):
        """
        :type A: List[int]
        :rtype: str
        """
        possibilities = sorted(list(itertools.permutations(sorted(A))), reverse=True)
        for i in possibilities:
            a, b, c, d = i
            hours = (a * 10 + b)
            minuites = (c * 10 + d)
            if hours < 24 and minuites < 60:
                return "{}{}:{}{}".format(a, b, c, d)
        return ""

File: leetcode.com/swift/DS_ALGO_PRAC.playground/Sources/swift/949_Largest_Time_for_Given_Digits.swift
class Solution {
    func largestTimeFromDigits(_ A: [Int]) -> String {
        var permutations = [[Int]]()
        let sortedA = A.sorted(by: >)
        for i in 0..<sortedA.count {
            for j in 0..<sortedA.count {
                if j == i {
                    continue
                }
                for k in 0..<sortedA.count {
                    if k == i || k == j {
                        continue
                    }
                    for l in 0..<sortedA.count {
                        if l == i || l == j || l == k {
                            continue
                        }
                        let permutation = [sortedA[i], sortedA[j], sortedA[k], sortedA[l]]
                        permutations.append(permutation)
                    }
                }
            }
        }
        
        for permutation in permutations {
            let (hour, minuites) = ((permutation[0]*10) + permutation[1], (permutation[2]*10) + permutation[3])
            if hour < 24 && minuites < 60 {
                var (hourStr, minuitesStr) = (String(hour), String(minuites))
                if hour < 10 {
                    hourStr = "0" + hourStr
                }
                if minuites < 10 {
                    minuitesStr = "0" + minuitesStr
                }
                return hourStr + ":" + minuitesStr
            }
         }
        return ""
    }
}
